signature/jws/jwt.go

```diff
@@ -0,0 +1,151 @@
+package jws
+
+import (
+	"crypto"
+	"crypto/x509"
+	"encoding/base64"
+	"fmt"
+
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/notaryproject/notation-core-go/signature"
+)
+
+// signingMethod is the interface for jwt.SigingMethod with additional method to
+// access certificate chain after calling Sign().
+type signingMethod interface {
+	jwt.SigningMethod
+
+	// CertificateChain returns the certificate chain.
+	//
+	// It should be called after calling Sign().
+	CertificateChain() ([]*x509.Certificate, error)
+
+	// PrivateKey returns the private key.
+	PrivateKey() crypto.PrivateKey
+}
+
+// remoteSigningMethod wraps the remote signer to be a SigningMethod
+type remoteSigningMethod struct {
+	signer    signature.Signer
+	certs     []*x509.Certificate
+	algorithm string
+}
+
+func newRemoteSigningMethod(signer signature.Signer) (signingMethod, error) {
+	algorithm, err := extractJwtAlgorithm(signer)
+	if err != nil {
+		return nil, err
+	}
+	return &remoteSigningMethod{
+		signer:    signer,
+		algorithm: algorithm,
+	}, nil
+}
+
+// Verify doesn't need to be implemented.
+func (s *remoteSigningMethod) Verify(signingString, signature string, key interface{}) error {
+	panic("not implemented")
```

Comment:
instead of using panic here, use fmt.Errorf just like func (s *localSigner) Sign(content []byte) ([]byte, []*x509.Certificate, error)

Reply:
Changed it to be an error.